llvm/include/llvm/CodeGen/TargetInstrInfo.h

```diff
@@ -2057,6 +2057,14 @@ class TargetInstrInfo : public MCInstrInfo {
         "Target didn't implement TargetInstrInfo::insertOutlinedCall!");
   }
 
+  /// Insert an architecture-specific instruction to clear a register.
+  virtual void buildClearRegister(Register Reg, MachineBasicBlock &MBB,
+                                  MachineBasicBlock::iterator Iter,
```

Comment:
In the interest of passing one less parameter, if `Iter`'s `->getParent() == MBB`, then consider just passing `Iter` (not `MBB`) and get `MBB` in `buildClearRegister` via `Iter->getParent()`. Not sure if that's the case though. If not, feel free to reject the suggestion.

Reply:
That could work, though I'm then a bit confused why `BuildMI` requires both the `MBB` and `Iter` values. My guess is that it has something to do with when `Iter` is at the end of the list?

If it's okay, I'd like to keep it like this for the time being. If it's possible to safely get the MBB from the iterator, then we should probably consider adding a `BuildMI` which takes only an iterator instead of both.

Reply:
That's fair, np.